hypervisor/lib/crypto/crypto_api.c

```diff
@@ -13,50 +13,46 @@ int32_t hkdf_sha256(uint8_t *out_key, size_t out_len,
 		const uint8_t *salt, size_t salt_len,
 		const uint8_t *info, size_t info_len)
 {
+	int32_t ret = 0;
 	const mbedtls_md_info_t *md;
 
 	/* salt and info can be NULL, others can't */
-	if (!out_key || !secret) {
-		return 0;
+	if ((out_key != NULL)  && (secret != NULL)) {
+		md = mbedtls_md_info_from_type(MBEDTLS_MD_SHA256);
+		if (md != NULL) {
+			ret = 1;
+			if (mbedtls_hkdf(md,
+					salt, salt_len,
+					secret, secret_len,
+					info, info_len,
+					out_key, out_len) != 0) {
```

Comment:
out_key, out_len) != 0) {
        ret = 1;

Reply:
changed the code